credentials/credentials.go

```diff
@@ -64,7 +64,7 @@ type Credentials interface {
 	// be used for timeout and cancellation.
 	// TODO(zhaoq): Define the set of the qualified keys instead of leaving
 	// it as an arbitrary string.
-	GetRequestMetadata(ctx context.Context) (map[string]string, error)
+	GetRequestMetadata(ctx context.Context, audience ...string) (map[string]string, error)
```

Comment:
I got notified this morning by the auth team to make this change and they said the design decision is firm. This is what java does https://github.com/google/google-auth-library-java/blob/master/credentials/java/com/google/auth/Credentials.java#L45. I borrowed some text and naming from there. I am still waiting for their update on designdoc so that I can make this better.